common/lib/xmodule/xmodule/course_module.py

```diff
@@ -1601,3 +1602,72 @@ def can_toggle_course_pacing(self):
           bool: False if the course has already started, True otherwise.
         """
         return datetime.now(UTC()) <= self.start
+
+
+class CourseSummaryDescriptor(object):
+    """
+    A lightweight course summary class, which parses split/mongo course information without loading
+    the course. It is used at cms for listing courses to global staff user.
+    """
+
+    def __init__(self, store, course_structure, course_locator):
+        """
+        Initialize and parse the `course_structure`
+        """
+        self.default_display_name = "Empty"
+        self.display_coursenumber = None
+        self.display_organization = None
+        self.course_info_fields = ['display_name', 'display_coursenumber', 'display_organization']
+
+        self.course_structure = course_structure
+
+        self.id = course_locator  # pylint: disable=invalid-name
+        self.location = course_locator.make_usage_key('course', 'course')
+
+        # Identify the store and extract course summary
+        if store == ModuleStoreEnum.Type.mongo:
+            self.extract_course_summary_for_mongo()
+        elif store == ModuleStoreEnum.Type.split:
+            self.extract_course_summary_split()
+
+        # If `display_name` isn't present in the course block, use the `default_display_name`. When you create a course
+        # from console, display_name isn't set (course block has no key `display_name`). "Empty" name is returned
+        # when we load the course. If display_name is not found, then apply "Empty" as display name.
+        if not hasattr(self, 'display_name'):
+            self.display_name = self.default_display_name
+
+    def extract_course_summary_for_mongo(self):
```

Comment:
This feels like it should be implemented in the modulestore layer, and that `CourseSummary` objects should be modulestore-agnostic. Can we instead have the the `self.course_structure` object always be structured the same when each modulestore produces it?